src/EFCore.PG/Scaffolding/Internal/NpgsqlDatabaseModelFactory.cs

```diff
@@ -511,191 +565,231 @@ NOT indisprimary
-                        .Where(ddr => ddr.GetValueOrDefault<char>("contype") == 'f'))
-                    {
-                        var fkName = foreignKeyRecord.GetValueOrDefault<string>("conname");
-                        var principalTableSchema = foreignKeyRecord.GetValueOrDefault<string>("fr_nspname");
-                        var principalTableName = foreignKeyRecord.GetValueOrDefault<string>("fr_relname");
-                        var onDeleteAction = foreignKeyRecord.GetValueOrDefault<char>("confdeltype");
-
-                        var principalTable = tables.FirstOrDefault(
-                                                 t => t.Schema == principalTableSchema
-                                                      && t.Name == principalTableName)
-                                             ?? tables.FirstOrDefault(
-                                                 t => t.Schema.Equals(principalTableSchema, StringComparison.OrdinalIgnoreCase)
-                                                      && t.Name.Equals(principalTableName, StringComparison.OrdinalIgnoreCase));
-
-                        if (principalTable == null)
-                        {
-                            _logger.ForeignKeyReferencesMissingPrincipalTableWarning(
-                                fkName,
-                                DisplayName(table.Schema, table.Name),
-                                DisplayName(principalTableSchema, principalTableName));
+                                primaryKey.Columns.Add(pkColumn);
+                            }
 
-                            continue;
+                            table.PrimaryKey = primaryKey;
+                            PkEnd: ;
                         }
 
-                        var foreignKey = new DatabaseForeignKey
+                        // Foreign keys
+                        foreach (var foreignKeyRecord in tableGroup.Where(ddr => ddr.GetValueOrDefault<char>("contype") == 'f'))
                         {
-                            Name = fkName,
-                            Table = table,
-                            PrincipalTable = principalTable,
-                            OnDelete = ConvertToReferentialAction(onDeleteAction)
-                        };
+                            var fkName = foreignKeyRecord.GetValueOrDefault<string>("conname");
+                            var principalTableSchema = foreignKeyRecord.GetValueOrDefault<string>("fr_nspname");
+                            var principalTableName = foreignKeyRecord.GetValueOrDefault<string>("fr_relname");
+                            var onDeleteAction = foreignKeyRecord.GetValueOrDefault<char>("confdeltype");
+
+                            var principalTable =
+                                tables.FirstOrDefault(
+                                    t => t.Schema == principalTableSchema
+                                         && t.Name == principalTableName)
```

Comment:
Move `t =>` to the same line where `FirstOrDefault` is located. The conditional and operator should be at the previous line end.

```csharp
var principalTable = tables.FirstOrDefault(t =>
    t.Schema == principalTableSchema && t.Name == principalTableName ||
    /* condition from the second FirstOrDefault*/);
```